src/Morpho.sol

```diff
@@ -432,18 +434,24 @@ contract Morpho is IMorphoStaticTyping {
 
     /// @inheritdoc IMorphoBase
     function setAuthorization(address authorized, bool newIsAuthorized) external {
+        require(newIsAuthorized != isAuthorized[msg.sender][authorized], ErrorsLib.ALREADY_SET);
+
         isAuthorized[msg.sender][authorized] = newIsAuthorized;
 
         emit EventsLib.SetAuthorization(msg.sender, msg.sender, authorized, newIsAuthorized);
     }
 
     /// @inheritdoc IMorphoBase
     function setAuthorizationWithSig(Authorization memory authorization, Signature calldata signature) external {
+        require(
+            authorization.isAuthorized != isAuthorized[authorization.authorizer][authorization.authorized],
+            ErrorsLib.ALREADY_SET
+        );
```

Comment:
This change is worrying me but it seems it's ok considering how bundlers are designed (possibility to not revert on signature frontrun)

Reply:
This change is not trivial at all because:
1. it makes it awkward to disable previously signed message. Assuming that you want to make sure that a previously signed message is not used, removing authorization with sig would revert, whereas the purpose of signing the message was to increase the nonce and disable other signatures. 
2. it makes it awkward for users to ensure that a target address is not authorized in the end. Assuming that there may be previously signed messages, the recommended action before this change was remove authorization with sig twice in different blocks, but this does not work anymore.

Those points at least require a comment explaining how to work around it: first give an authorization with sig to another dummy address (such as address(0)). For point 2, the user should also disable the target address afterwards (with or without sig).

Because I don't see a clear upside to this change, reverting it seems simpler than having a workaround and documenting it

Reply:
> Because I don't see a clear upside to this change, reverting it seems simpler than having a workaround and documenting it

I agree

Reply:
I assume that this change has been reverted at some point in another PR? Because it's not present in the `post-cantina` branch.

I understand your point, but I think that it's safer to not have it because it allows you to "burn" a signed message if you need to (consume the nonce) even if it would make no change.

For example, 

- user Alice is authed
- you have a signed message with a long deadline that would re-auth Alice

you don't trust her anymore, what you would do is to

- execute the `setAuthorizationWithSig(...)` even if it does nothing (because Alice already has the auth)
- execute `setAuthorization(alice, false)` to remove auth from her


it's true that you could also do this

- generate a new signed message with the same nonce that would remove the auth from Alice and execute it.

You would probably need to monitor reverts because of front running anyway.

Reply:
yes it has been reverted by https://github.com/morpho-org/morpho-blue/pull/652

Reply:
>I understand your point, but I think that it's safer to not have it because it allows you to "burn" a signed message if you need to (consume the nonce) even if it would make no change.

not that this was possible why for example enabling address zero (or any other dumb address)